Makefile

```diff
@@ -25,10 +25,14 @@ hubble:
 	$(GO_BUILD) $(if $(GO_TAGS),-tags $(GO_TAGS)) -ldflags "-w -s -X 'github.com/cilium/hubble/pkg.GitBranch=${GIT_BRANCH}' -X 'github.com/cilium/hubble/pkg.GitHash=$(GIT_HASH)' -X 'github.com/cilium/hubble/pkg.Version=${VERSION}'" -o $(TARGET)
 
 release:
-	docker run --env "RELEASE_UID=$(RELEASE_UID)" --env "RELEASE_GID=$(RELEASE_GID)" --rm --workdir /hubble --volume `pwd`:/hubble docker.io/library/golang:1.18.3-alpine3.16 \
-		sh -c "apk add --no-cache make git && make local-release"
+	docker run --rm --workdir /hubble --volume `pwd`:/hubble docker.io/library/golang:1.18.3-alpine3.16 \
+		sh -c "apk add --no-cache make git && \
+			addgroup -g $(RELEASE_GID) release && \
+			adduser -u $(RELEASE_UID) -D -G release release && \
+			su release -c 'make local-release'"
```

Comment:
FYI, we've had the same issue in cilium-cli and resolved it slightly differently: https://github.com/cilium/cilium-cli/commit/023ccfb8466e2c649230b5cf6c98260bee85c863

Not sure whether that approach is preferable though :thinking:

Reply:
Thanks! Yeah, I was aware of `safe.directory`, but since we still need to chmod/chown the built binaries, I figured we might as well switch the user instead.

Reply:
Thanks, that's a good point! We should probably change to the approach taken in this PR for cilium-cli as well.

Reply:
FWIW, I've opened cilium/cilium-cli#945 to switch to this PR's approach in cilium-cli as well. Thanks again.